### DISCUSSION TITLE
Modify the AXI interface of FINN Stitched IP

### DISCUSSION THREAD

**Discussion by laochonlam** [Xilinx/finn#522] [Q&A] [answered]
Hi FINN developers,

I want to integrate FINN Stitched IP with my Vivado design. However, I found out that the FINN IP's AXI-Stream interface doesn't expose the `tlast` and `tkeep` interfaces. May I know what is the best way to modify the interfaces in order to integrate them into my Vivado design?

Thanks,
Lam

**Comment by heborras** [comment, marked as answer]
Hi @laochonlam,

exposing `tlast` is indeed possible using the `InsertTLastMarker` transformation: https://github.com/Xilinx/finn/blob/d1cc9cf94f1c33354cc169c5a6517314d0e94e3b/src/finn/transformation/fpgadataflow/insert_tlastmarker.py#L38

This must be done after the FIFO depths have been set. A bit of a convoluted example of this use-case can be found here: https://github.com/fastmachinelearning/finn-examples/blob/e36659647cc799dd44c7bd12b8be20df487fe55c/build/kws-pynqz-axidma/build.py#L51

However, as far as I know FINN does not implement the `tkeep` interface.

Cheers,
Hendrik

**Reply by laochonlam** [reply]
Thanks for letting me know it!

**Reply by JPPalacios** [reply]
Hi @heborras @laochonlam , 

Is this still applicable? I am doing something similar but working on the latest version of FINN. Thanks!

**Reply by heborras** [reply]
Hi @JPPalacios 
this appears to still be applicable, as far as I can tell from a quick look at the current FINN source code on main, here: https://github.com/Xilinx/finn/blob/9d299689f2ec0895f208b8bfe3bcdcf6f450181a/src/finn/transformation/fpgadataflow/insert_tlastmarker.py#L37
As in the transformation still exists and is still supposed to do the same thing. Note though, that I haven't used the feature in recent years, so your mileage may vary.
